runtime/src/main/java/org/corfudb/security/tls/ReloadableTrustManager.java

```diff
@@ -35,16 +34,43 @@ public ReloadableTrustManager(TrustStoreConfig trustStoreConfig) throws SSLExcep
 
     @Override
     public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        if (isCertExpiryCheckEnabled()) {
+            for (X509Certificate cert : chain) {
+                cert.checkValidity();
+            }
+        } else {
+            logCertExpiryCheck();
+        }
+
         reloadTrustStoreWrapper();
         trustManager.checkClientTrusted(chain, authType);
     }
 
     @Override
     public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        if (isCertExpiryCheckEnabled()) {
+            for (X509Certificate cert : chain) {
+                cert.checkValidity();
+            }
+        } else {
+            logCertExpiryCheck();
+        }
+
```

Comment:
Can you consolidate this block and the one above into the same function -- i.e. `checkCertValidity`?

Reply:
+1